File: frontend/src/types/audit.ts
export type AuditCategory =
  | 'scrape_run'
  | 'resume_generation'
  | 'ats_score'
  | 'application_submitted'
  | 'email_sent'
  | 'captcha_blocked';

export type AuditStatus = 'success' | 'warning' | 'error' | 'captcha_blocked';

export interface AuditLogEntry {
  id: number;
  user_id: number;
  category: AuditCategory;
  action: string;
  details?: string;
  status: AuditStatus;
  timestamp: string;
}

File: backend/models/audit.py
from sqlalchemy import Column, Integer, String, Text, DateTime, JSON
from datetime import datetime
from backend.database import Base

class AuditLog(Base):
    __tablename__ = "audit_logs"

    id = Column(Integer, primary_key=True, index=True)
    action = Column(String(255), index=True)  # e.g., "DISCOVERY_SCAN", "TAILOR_RESUME", "ATS_SCORE", "COLD_EMAIL_SENT"
    source = Column(String(100), default="System")  # e.g., "Celery Worker", "User UI", "Scraper"
    status = Column(String(50), default="SUCCESS")  # "SUCCESS", "WARNING", "ERROR", "BLOCKED"
    details = Column(Text, default="")
    metadata_json = Column(JSON, default=dict)
    
    timestamp = Column(DateTime, default=datetime.utcnow, index=True)
